### ISSUE TITLE
[2.2.0 UHT Generator] PublicDependencyModuleNames list is not populated correctly

### ISSUE THREAD

**Issue by Buckminsterfullerene02** [UE4SS-RE/RE-UE4SS#36]
In the <name>.Build.cs file for the project, the `PublicDependencyModuleNames` array is not generated properly anymore.

For example, old version:
```cpp
PublicDependencyModuleNames.AddRange(new string[] {
            "AIModule",
            "AnimGraphRuntime",
            "AnimationCore",
            "AnimationSharing",
            "AudioMixer",
            "Chaos",
            "Core",
            "CoreUObject",
            "DLSSBlueprint",
            "DSTelemetry",
            "DeveloperSettings",
            "Engine",
            "FSDEngine",
            "GameplayTags",
            "GameplayTasks",
            "InputCore",
            "LevelSequence",
            "MediaAssets",
            "MediaUtils",
            "MovieScene",
            "MovieSceneTracks",
            "NavigationSystem",
            "Niagara",
            "NiagaraCore",
            "NiagaraShader",
            "OnlineSubsystem",
            "OnlineSubsystemUtils",
            "PhysicsCore",
            "PropertyPath",
            "Slate",
            "SlateCore",
            "UMG",
        });
```

New version:
```cpp
PublicDependencyModuleNames.AddRange(new string[] {
            "AIModule",
            "Core",
            "CoreUObject",
            "Engine",
            "FSDEngine",
            "GameplayTasks",
            "NavigationSystem",
            "Niagara",
            "PhysicsCore",
            "UMG",
        });
```

1. Use following settings:
```
IgnoreAllCoreEngineModules = 0
IgnoreEngineAndCoreUObject = 1
MakeAllFunctionsBlueprintCallable = 1
MakeAllPropertyBlueprintsReadWrite = 1
MakeEnumClassesBlueprintType = 1
```
2. UHT Dump
3. Fix all normal errors
4. Watch as your world burns with many headers missing errors

This issue is almost certianly cost by [this line of code](https://github.com/UE4SS-RE/RE-UE4SS/commit/eed11361a09c17c27366590f16f9cd6cff5917f7#diff-f2d1c353a4bdf4ce653bc93b980513944149dc55cbd2d6e5f220d6ea1c6a9692R1384) in #33  

According to the changelogs, this is only meant to be checked when the engine dumping is enabled, but the current code does not appear to actually check the state of that setting. 

@praydog can you confirm/look into this? Obviously I don't want to undo whatever was causing the crash for you previously.

**Comment by Buckminsterfullerene02** [comment]
Looks like this issue is also the cause of hundreds of errors stemming from the plugins' generated Build.cs files missing modules too.
![image](https://user-images.githubusercontent.com/84156063/212705509-0df55a5e-26b8-4ca7-bfc8-8988fa2a628b.png)

**Comment by praydog** [comment]
#37 Should fix. Sorry about that.

**Comment by narknon** [comment]
Fixed via #37 and hotfix released as v2.2.1-Hotfix
